### PR TITLE
Permission module additions

### PR DESCRIPTION
**Pull request by Dubjunk**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| Needs wipe? | no |
| Fixed issues | #571  |

### Description:
This PR provides some smaller additions for the permission module.
A fresh save requires now an admin login to setup the basic permissions like `commander` / `subcommander` access.
Additionally the `Build Menu` and `Logistic Station` access can now be handled via permission dialog.

### Content:
- [x] `Build` permission
- [x] `Logistic Station` permission
- [x] `Commander` & `Sub-Commander` permission
- [x] Additions to the `registerPlayer` function
- [x] CBA settings for permission dialog access

### Successfully tested on:
- [X] Local MP Vanilla
- [X] Dedicated MP Vanilla

### Compatibility checked with:
* ACE

### CODE REVIEW

File: Missionframework/modules/11_permission/fnc/fn_permission_setupPlayerActions.sqf [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 // Actions avalible LOCALLY to player
 // Permission action
-private _permissionCondition = '_target isEqualTo _originalTarget && serverCommandAvailable "#kick"';
+private _permissionCondition = '_target isEqualTo _originalTarget && (serverCommandAvailable "#kick" || (KPLIB_param_permissionCommander && (str player) isEqualTo "KPLIB_eden_commander") || KPLIB_param_permissionSubCommander && (str player) isEqualTo "KPLIB_eden_subCommander")';
```
Comment (Wyqer) [inline Missionframework/modules/11_permission/fnc/fn_permission_setupPlayerActions.sqf]:
```suggestion
private _permissionCondition = '
    _target isEqualTo _originalTarget &&
    (serverCommandAvailable "#kick" ||
        (KPLIB_param_permissionCommander && (str player) isEqualTo "KPLIB_eden_commander") ||
        (KPLIB_param_permissionSubCommander && (str player) isEqualTo "KPLIB_eden_subCommander")
)';
```
Just a suggestion for a formatting which is easier to read for all valid cases (admin, commander or sub commander)

File: Missionframework/modules/15_build/fnc/fn_build_setupPlayerActions.sqf [hunk 2]
```diff
@@ -20,7 +20,7 @@
 // Actions avalible LOCALLY to player
 if(hasInterface) then {
     // Build action
-    private _buildCondition = '_target == _originalTarget && !(_originalTarget getVariable ["KPLIB_fob", ""] in ["", "KPLIB_eden_startbase_marker"])';
+    private _buildCondition = '_target == _originalTarget && !(_originalTarget getVariable ["KPLIB_fob", ""] in ["", "KPLIB_eden_startbase_marker"]) && [player, "Build"] call KPLIB_fnc_permission_getPermission';
```
Comment (Wyqer) [inline Missionframework/modules/15_build/fnc/fn_build_setupPlayerActions.sqf]:
```suggestion
    private _buildCondition = '
        (_target == _originalTarget) &&
        !(_originalTarget getVariable ["KPLIB_fob", ""] in ["", "KPLIB_eden_startbase_marker"]) &&
        ([player, "Build"] call KPLIB_fnc_permission_getPermission)
    ';
```
It's already fine, but after the first suggestion I thought these "longer &&/|| conditions" could be formatted like that in the future or replace as soon as we see some.

File: Missionframework/modules/11_permission/fnc/fn_permission_setupPlayerActions.sqf (hunk 1 above)
Comment (Dubjunk) [reply Missionframework/modules/11_permission/fnc/fn_permission_setupPlayerActions.sqf]: The last bracket looks a little bit weird maybe we should bring it in a line with the bracket before `serverCommandAvailable` and push the command to the next line

Comment (Wyqer) [reply Missionframework/modules/11_permission/fnc/fn_permission_setupPlayerActions.sqf]: Like you prefer it. Not that important in the end.

**Pull request merged**